lib/fuel-rest.js

```diff
@@ -97,18 +97,24 @@ FuelRest.prototype._processRequest = function(options, callback) {
 			return;
 		}
 
-		retry       = options.retry || false;
 		authOptions = clone(options.auth);
 
-		delete options.retry;
-		delete options.auth;
-
 		options.uri     = helpers.resolveUri(this.origin, options.uri);
 		options.headers = merge({}, this.defaultHeaders, options.headers);
-		options.headers.Authorization = options.headers.Authorization || 'Bearer ' + authResponse.accessToken;
+
+		if(options.headers.Authorization) {
+			retry = false;
```

Comment:
Based on that new realization, we should probably just `var retry = false` at the top, and these two blocks can just become one:
```js
if(!options.headers.Authorization) {
    options.headers.Authorization = 'Bearer ' + authResponse.accessToken;
    retry = options.retry || false;
}
```